D/D.sublime-syntax

```diff
@@ -8,339 +8,2032 @@ file_extensions:
+      set:
+        - match: '\b({{name}})\b'
+          scope: entity.name.enum.d
+          set: enum-assignment
+        - include: not-whitespace-illegal-pop
+    - include: not-whitespace-illegal-pop
+
+  anonymous-enum-body:
+    - match: '{'
+      scope: punctuation.section.block.begin.d
+      set:
+        - match: '}'
+          scope: punctuation.section.block.end.d
+          pop: true
+        - include: attribute-in
+        - match: '(?=\b({{name}})\s*(=|,|}|$))'
+          push: enum-member
+        - match: '{{type_look_ahead}}'
+          push: [enum-member, type]
+    - include: not-whitespace-illegal
+
+  enum-member:
+    - match: '\b({{name}})\b'
+      scope: entity.name.constant.d
+      set:
+        - match: '='
+          scope: keyword.operator.assignment.d
+          set: [enum-member-after, value]
+        - match: ','
+          scope: punctuation.separator.sequence.d
+          pop: true
+        - match: '(?=})'
+          pop: true
+        - include: not-whitespace-illegal
+    - include: not-whitespace-illegal
+  enum-member-after:
+    - match: ','
+      scope: punctuation.separator.sequence.d
+      pop: true
+    - match: '(?=})'
+      pop: true
+    - include: not-whitespace-illegal-pop
+
+  conditional-declaration-in:
+    - match: '\b(version)\b'
+      scope: keyword.control.conditional.d
+      push: [optional-block-statement-or-label, version-condition]
+    - match: '\b(debug)\b'
+      scope: keyword.control.conditional.d
+      push: [optional-block-statement-or-label, debug-condition]
+
+  version-condition:
+    - match: '\('
+      scope: punctuation.section.parens.begin.d
+      set:
+        - match: '\b(unittest|assert)\b'
+          scope: keyword.d
+          set: version-condition-end
+        - match: '\b({{name}})\b'
+          scope: constant.other.d
+          set: version-condition-end
+        - match: '{{integer_lookahead}}'
+          set: [version-condition-end, integer]
+        - include: not-whitespace-illegal
+    - include: not-whitespace-illegal
+  version-condition-end:
+    - match: '\)'
+      scope: punctuation.section.parens.end.d
+      pop: true
+    - include: not-whitespace-illegal
+
+  debug-condition:
+    - match: '\('
+      scope: punctuation.section.parens.begin.d
+      set:
+        - match: '\b({{name}})\b'
+          scope: constant.other.d
+          set: version-condition-end
+        - match: '{{integer_lookahead}}'
+          set: [version-condition-end, integer]
+        - include: not-whitespace-illegal
+    - match: '(?=\S)'
+      pop: true
+
+  static-foreach-in:
+    - match: '\b(static)\s+(foreach|foreach_reverse)\b'
       captures:
-        1: punctuation.definition.comment.d
-  constant_placeholder:
-    - match: '(?i:%(\([a-z_]+\))?#?0?\-?[ ]?\+?([0-9]*|\*)(\.([0-9]*|\*))?[hL]?[a-z%])'
-      scope: constant.other.placeholder.d
-  function_assertions:
-    - match: '(in|out)\s*(?=\{)'
+        1: keyword.control.loop.d
```

Comment:
This seems like an unusual scope for `static`, but I am not familiar with its usage.

Reply:
`static` in this case means "compile time" (which has some other implications), but in D these kinds of statements are specifically referred to as "static if", "static foreach" and "static assert". Because of that I felt it better to have it appear more as a 2 word keyword than a "modifier" and a keyword, as this fits its use and meaning better.

Reply:
It might be a bit weird if they type `static` it highlights as an attribute but finish with `static foreach` it changes to a modifier. Not sure on this.